### PR TITLE
Title case fixes

### PR DESCRIPTION
**Pull request by tmorrell**
The main reason for the PR is to fix title case for titles with multiple capital letters (e.g. Three new iron-phosphate minerals from the El Ali iron meteorite, Somalia: Elaliite, Fe²⁺₈Fe³⁺ (PO₄)O₈; elkinstantonite, Fe₄(PO₄)₂O; and olsenite, KFe₄(PO₄)₃ was previously rendered as Three New Iron-phosphate Minerals from the El Ali Iron Meteorite, Somalia: Elaliite, Fe²⁺₈fe³⁺ (PO₄)O₈; Elkinstantonite, Fe₄(po₄)₂o; and Olsenite, Kfe₄(po₄)₃). The previous version lowercased a lot more that is needed for title case. 

This will fix [textcase_TitleCaseWithCleverBrandName.txt](https://github.com/citation-style-language/test-suite/blob/master/processor-tests/humans/textcase_TitleCaseWithCleverBrandName.txt), which will show up on the failed list once https://github.com/brechtm/citeproc-py/pull/139 is merged.

https://github.com/brechtm/citeproc-py/pull/138 was needed for me to run tests locally.

### CODE REVIEW

**Comment by zepinglee** [comment]
> Oddly there issn't a csl test case that covers this specific situation, but this change hasn't impacted the test coverage of citeproc-py.

Actually there is a test case [textcase_TitleCaseWithCleverBrandName.txt](https://github.com/citation-style-language/test-suite/blob/master/processor-tests/humans/textcase_TitleCaseWithCleverBrandName.txt) but it's not included in the current test-suite commit.

https://github.com/brechtm/citeproc-py/blob/fef20586ddca0fac4d0c301119a871625a21926f/tests/citeproc-test.py#L19

File: citeproc/model.py [hunk 1]
```diff
@@ -373,13 +373,14 @@ def case(self, text, language=None):
...
                 output = []
+                # : gurantees that the first word is capitalized
                 prev = ':'
                 for word in text.words():
-                    if not text.isupper() and not word.isupper():
+                    if (str(word) not in self._stop_words or
+                        prev in (':', '.')):
+                        word = word.capitalize_first()
+                    else:
                         word = word.soft_lower()
```
Comment (zepinglee) [inline citeproc/model.py:383]: The v1.0.1 [spec](https://docs.citationstyles.org/en/v1.0.1/specification.html#title-case-conversion) says:

> For lower or mixed case strings, the first character of each lowercase word is capitalized. The case of words in mixed or uppercase stays the same.

I suppose we should ensure the `word` is in lowercase and drop the `else` branch.

```python
                    if word.islower() and (str(word) not in self._stop_words or
                        prev in (':', '.')):
                        word = word.capitalize_first()
```

File: setup.py [hunk 2]
```diff
@@ -88,8 +60,8 @@ def run(self):
 
 setup(
     name='citeproc-py',
-    version=__version__,
-    cmdclass = dict(build_py=custom_build_py, develop=custom_develop),
+    version=versioneer.get_version(),
+    cmdclass=versioneer.get_cmdclass({'build_py':custom_build_py,'develop':custom_develop}),
```
Comment (yarikoptic) [inline setup.py:64]: lint
```suggestion
    cmdclass=versioneer.get_cmdclass({'build_py': custom_build_py, 'develop': custom_develop}),
```

File: citeproc/model.py [hunk 3]
```diff
@@ -373,13 +373,14 @@ def case(self, text, language=None):
...
             elif text_case == 'title':
                 output = []
+                # : gurantees that the first word is capitalized
                 prev = ':'
                 for word in text.words():
-                    if not text.isupper() and not word.isupper():
+                    if (str(word) not in self._stop_words or
+                        prev in (':', '.')):
+                        word = word.capitalize_first()
+                    else:
```
Comment (yarikoptic) [inline citeproc/model.py:382]: I am just a fly on the wall here, but I would recommend to
- separate versioneer addition into a separate PR
- accompany PR with code fixes with a test demonstrating/solidifying the need for hte fix

Comment (yarikoptic) [reply citeproc/model.py:382]: FWIW fixing up/extending testing in 
- #137 
so hopefully it would soon gets triggered again on PRs here too

File: setup.py (hunk 2 above)
Comment (tmorrell) [reply setup.py]: Thanks! Implemented in versioneer PR

File: citeproc/model.py (hunk 3 above)
Comment (tmorrell) [reply citeproc/model.py]: Thanks, I've split everything up into independent PRs, and the updated tests will show what's fixed by this PR. I think you need to approve the workflow for the CI to run.

**Comment by yarikoptic** [comment]
> This will fix [textcase_TitleCaseWithCleverBrandName.txt](https://github.com/citation-style-language/test-suite/blob/master/processor-tests/humans/textcase_TitleCaseWithCleverBrandName.txt), which will show up on the failed list once #139 is merged.

I am a little confused -- shouldn't then CI run on #139 fail ? it seems to be happy without any fixes

**Comment by tmorrell** [comment]
> > This will fix [textcase_TitleCaseWithCleverBrandName.txt](https://github.com/citation-style-language/test-suite/blob/master/processor-tests/humans/textcase_TitleCaseWithCleverBrandName.txt), which will show up on the failed list once #139 is merged.
> 
> I am a little confused -- shouldn't then CI run on #139 fail ? it seems to be happy without any fixes

I added [textcase_TitleCaseWithCleverBrandName.txt](https://github.com/citation-style-language/test-suite/blob/master/processor-tests/humans/textcase_TitleCaseWithCleverBrandName.txt), to the failed tests file https://github.com/brechtm/citeproc-py/pull/139/files#diff-5561a0cbd824d26d44da1b4df2b3bbb9749a960d95b3e8b932974500c30e2862 so the CI will pass. Once #139 is merged I then remove it from the failed tests file in this PR.

**Comment by yarikoptic** [comment]
sorry -- I have managed to miss that. Thank you for the explanation, let's proceed there.

**Comment by tmorrell** [comment]
Thanks! This is good to go with the fixed test now.
